### ISSUE TITLE
Investigate gpkg pivot view for arcgis

### ISSUE THREAD

**Issue by KellyMWhitehead** [Riverscapes/riverscapes-tools#265]
Test to make sure arcgis correctly displays a spatial layer based on a geopackage pivot table.

**Comment by philipbaileynar** [comment]
TLDR

1. a SQLite spatial view using `CASE` statements to fake a pivot table does show up in ArcGIS and can be added to the map.
2. Unfortunately the pivoted columns (read "calculated columns") don't appear in ArcGIS 10.6.1.

More Detail...

I took a BRAT project and added two new tables.

Attributes:

<img width="173" alt="Screen Shot 2021-03-24 at 4 11 38 PM" src="https://user-images.githubusercontent.com/8645379/112395441-9f0af300-8cbb-11eb-9ecc-50795275ef74.png">

and Attributes Values:

<img width="220" alt="Screen Shot 2021-03-24 at 4 12 05 PM" src="https://user-images.githubusercontent.com/8645379/112395469-af22d280-8cbb-11eb-82fb-8eb9ff4e704b.png">

The latter is a junction table between the feature class table ReachGeometry and Attributes.

I then created a view that joins all three tables together. And another view that essentially pivots the data using CASE statements, a technique that I've seen written in several places (including [here](https://stackoverflow.com/questions/1237068/how-to-pivot-in-sqlite-or-i-e-select-in-wide-format-a-table-stored-in-long-form)).

```sql
SELECT ReachId,
       geom,
       Sum(CASE WHEN Name = 'attribute 1' THEN Value END) AS Attribute1,
       Sum(CASE WHEN NAME = 'attribute 2' THEN Value END) AS Attribute2
  FROM vwAttributes
 GROUP BY ReachID, geom
```

Note that I did indeed "register" this view in the gpkg_contents table.

The good news is that the layer displays in ArcGIS 10.6.1. The bad news is that none of the columns that use the CASE statement appear as attributes:

<img width="539" alt="Screen Shot 2021-03-24 at 4 18 08 PM" src="https://user-images.githubusercontent.com/8645379/112395918-87803a00-8cbc-11eb-9f80-ca6e09bf050c.png">

**Comment by philipbaileynar** [comment]
Next I tried the `LEFT JOIN` approach described in the linked stack exchange post above. I was more hopeful because this doesn't produce any calculated columns. Still didn't work!

```sql
SELECT R.ReachID,
       geom,
       A.Value Attribute1,
       B.Value Attribute2
  FROM ReachGeometry R
       LEFT JOIN
       AttributeValues A ON R.ReachID = A.ReachID AND 
                            A.AttributeID = 1
       LEFT JOIN
       AttributeValues B ON R.ReachID = B.ReachID AND 
                            B.AttributeID = 2
 WHERE Attribute1 IS NOT NULL AND 
       Attribute2 IS NOT NULL
```

**Comment by philipbaileynar** [comment]
Reading the ArcGIS help on [SQLite data types](https://desktop.arcgis.com/en/arcmap/10.3/manage-data/databases/dbms-data-types-supported.htm#ESRI_SECTION1_E1310ADFB340464485BA2D2D167C9AE4) didn't help either.
